### ISSUE TITLE
Nan-style require() for including headers

### ISSUE THREAD

**Issue by mapsam** [mapbox/node-cpp-skel#21]
Same as https://github.com/mapbox/hpp-skel/issues/7

Let's take a cue from Nan and have a `include_dirs.js` file that allows compliers to use the following syntax when including libs and headers:

```
'<!(node -e \'require("hpp-skel")\')'
```

This is super helpful when we run into multiple versions of header files when things are installed with npm.

cc @springmeyer @GretaCB

**Comment by GretaCB** [comment]
Next actions:
- Document how to add new dependencies to the skel. Using `include_dir` trick is one thing to know for this.
- Publish hpp-skel to npm and mason (header-only cpp libs should be published to npm if they're ever going to be used by a node module. If header-only cpp libs are going to be pulled into a pure C++ project, then publish with mason). **Link to [hpp-skel docs about publishing](https://github.com/mapbox/hpp-skel#how-to-publishinclude-a-c-header-library) in cpp and node-cpp repos**
- Use hpp-skel as dependency in node-cpp-skel as part of `include_dir`

**Comment by springmeyer** [comment]
Our thinking has changed here, so no action is needed. I will recap:

 - Previously we packaged C++ header only libs in both mason and npm
 - Those npm packaged libs could be referenced in package.json, but it was hard to dynamically know their include paths for compiling against
 - The `include_dirs.js` trick was the solution.

However now we've moved away from packaging C++ libs in npm and only are targeting mason per https://github.com/mapbox/cpp/issues/20#issuecomment-312389430.

So, any deps of node-cpp-skel or apps that use it will use mason for bringing in C++ headers. I've added an example of this in https://github.com/mapbox/node-cpp-skel/pull/43
